src/portal/mod.rs

```diff
@@ -104,48 +107,65 @@ impl Keyring {
             }
         };
 
-        let key = keyring.derive_key(secret);
-
         Ok(Self {
             keyring: Mutex::new(keyring),
             path: path.as_ref().to_path_buf(),
             mtime: Mutex::new(mtime),
-            key,
+            secret: Zeroizing::new(secret.to_vec()),
+            key: OnceCell::new(),
         })
     }
 
+    /// Force to derive key
+    ///
+    /// Usually the key is only derived when required. This usually happens
+    /// when optaining in element. With this function it will already be
+    /// derived and cached beforehand.
+    pub async fn force_dervice_key(&self) {
```

Comment:
Not really sure we want this api to be public?

Reply:
At this point, I would make KeyExt part of the unstable feature so you could call it without having to expose this function

Reply:
I don't need it. But some apps might want to lookup a key quickly while running and therefore might want to do the derivation on startup. That was my idea behind this function.

Reply:
I see. Well unless people need more APIs I wouldn't add them because removing them is harder :P